File: codegen/results/v002/gemma4_26b_think/20260411_182157/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * This script reads a CSV file from 'input/data.csv' and converts its content
 * into a JSON array of objects. Each object represents a row from the CSV,
 * with fields: Name (string), Age (integer), Email (string), and Score (float).
 */

function main() {
    const inputPath = path.join('input', 'data.csv');

    try {
        // Check if the file exists before attempting to read it.
        if (!fs.existsSync(inputPath)) {
            // If the file is missing, we output an empty array to ensure the output remains valid JSON.
            process.stdout.write('[]\n');
            return;
        }

        // Read the file content.
        const content = fs.readFileSync(inputPath, 'utf-8');

        // Split content into lines and remove empty lines (e.g., trailing newline).
        const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');

        // If there's only a header or no content, return an empty array.
        if (lines.length <= 1) {
            process.stdout.write('[]\n');
            return;
        }

        // The first line is the header, so we skip it.
        const dataRows = lines.slice(1);

        // Transform each data row into an object.
        const result = dataRows.map(row => {
            // Split the row by comma and trim whitespace from each value.
            const columns = row.split(',').map(col => col.trim());

            // Destructure columns based on the expected schema.
            // Index 0: Name, Index 1: Age, Index 2: Email, Index 3: Score
            const name = columns[0];
            const age = parseInt(columns[1], 10);
            const email = columns[2];
            const score = parseFloat(columns[3]);

            return {
                Name: name,
                Age: isNaN(age) ? 0 : age,
                Email: email,
                Score: isNaN(score) ? 0.0 : score
            };
        });

        // Output the resulting array as a JSON string to stdout.
        // Using null and 2 for pretty-printing, but it can be JSON.stringify(result) for compact.
        process.stdout.write(JSON.stringify(result, null, 2) + '\n');

    } catch (error) {
        // In case of a critical error, we output an empty array to maintain valid JSON output constraint.
        process.stdout.write('[]\n');
    }
}

// Execute the script.
main();

File: codegen/results/v002/gemma4_26b_think/20260411_182157/python/001_csv_to_json/solution.py
import csv
import json
import sys

def main():
    """
    Reads a CSV file from 'input/data.csv', processes the rows into
    a list of dictionaries with specific types, and outputs the 
    result as a JSON array to stdout.
    """
    input_file_path = 'input/data.csv'
    output_data = []

    try:
        with open(input_file_path, mode='r', encoding='utf-8') as csvfile:
            # Use DictReader to automatically map the header row to dictionary keys
            reader = csv.DictReader(csvfile)
            
            for row in reader:
                # Create the object with the required type conversions:
                # Name: string, Age: integer, Email: string, Score: float
                try:
                    record = {
                        "Name": str(row["Name"]),
                        "Age": int(row["Age"]),
                        "Email": str(row["Email"]),
                        "Score": float(row["Score"])
                    }
                    output_data.append(record)
                except (ValueError, KeyError) as e:
                    # If a row is malformed or missing a key, we skip it to maintain 
                    # the validity of the final JSON array output.
                    continue
        
        # Serialize the list of dictionaries to a JSON string
        # We output to stdout as requested.
        json_output = json.dumps(output_data, indent=4)
        sys.stdout.write(json_output + '\n')

    except FileNotFoundError:
        # If the file is not found, we output nothing to adhere to the 
        # "Do not output anything else" constraint.
        pass
    except Exception:
        # Catch-all for unexpected errors to ensure no non-JSON text is printed.
        pass

if __name__ == "__main__":
    main()